IGNITE-17347 Add port parameters to Ignite3 CLI node start command

valepakh opened the pull request:
This will add --port, --rest-port and --join parameters to simplify starting nodes especially for local testing.

https://issues.apache.org/jira/browse/IGNITE-17347

PakhomovAlexander commented on modules/cli/src/main/java/org/apache/ignite/cli/deprecated/spec/NodeCommandSpec.java:
```diff
@@ -100,12 +123,36 @@ public Integer call() {
+                return null;
+            }
+            // port is required
+            StringBuilder sb = new StringBuilder();
```
I think it is better to use `Config` class from `typesafe` lib to construct a configuration representation.

PakhomovAlexander commented on modules/cli/src/main/java/org/apache/ignite/cli/deprecated/spec/NodeCommandSpec.java:
```diff
@@ -62,12 +65,31 @@ public static class StartNodeCommandSpec extends BaseCommand implements Callable
+        private ConfigOptions configOptions;
+
+        private static class ConfigOptions {
+            @ArgGroup(exclusive = false)
```
Could you add an unit test to verify that it is truly exclusive?

sk0x50 commented on modules/cli/src/main/java/org/apache/ignite/cli/deprecated/builtins/node/NodeManager.java:
```diff
@@ -136,6 +137,9 @@ public RunningNode start(
                 cmdArgs.add("--config");
                 cmdArgs.add(srvCfg.toAbsolutePath().toString());
+            } else if (srvCfgStr != null) {
+                cmdArgs.add("--configStr");
```
`--configStr` does not look descriptive to be honest :) By the way, do we use camel case for commands? Perhaps it should be named `config-str`, for instance see at `work-dir`.

valepakh replied:
`--config-file` and `--config-string` maybe?

sk0x50 replied:
Perhaps, `config-overrides` would cleaner. What do you think?

sk0x50 replied:
Looks like I was wrong. My last comment does not make sense. Please ignore.

sk0x50 commented on modules/cli/src/main/java/org/apache/ignite/cli/deprecated/spec/NodeCommandSpec.java:
```diff
@@ -102,12 +130,32 @@ public Integer call() {
+            return configOptions != null ? configOptions.configPath : null;
+        }
+
+        private String getConfigStr() {
```
Well, looks like these parameters (port, rest-port etc) cannot be provided with a configuration file. Right?
It seems to me, that ability to start two nodes with the same configuration file and overridden value of `port` would be very convenient. I mean the following:
`node start nodeName --config \path_to_configuration_file --port 12345`

What do you think?

valepakh replied:
Yes I think this is possible and will be helpful.

valepakh commented:
I changed CLI runner parameters to `--config-path` and `--config-string` and implemented overriding config file with parameters.
Also I improved the IgniteCliInterfaceTest so that the mockserver only starts once, this significantly improved performance when running on my local machine in IDEA.
